transports/upgrades/sqlserver-unicode-headers.md

```diff
@@ -0,0 +1,64 @@
+---
+title: SQL Server Transport Upgrade - Supporting Unicode in headers
+summary: Instructions on how to add support for Unicode characters in message headers
+component: SqlTransport
+reviewed: 2017-07-17
+isUpgradeGuide: true
+upgradeGuideCoreVersions:
+ - 5
+ - 6
+---
+
+
+## Summary
+
+This document explains how to patch a system to [support Unicode characters in message headers using the SQL Server transport](https://github.com/Particular/NServiceBus.SqlServer/issues/340). The issue may cause data loss when using header names or values that contain Unicode characters.
+
+
+## Compatibility
+
+This issue has been resolved in the following patch versions of the [SQL Server transport](/transports/sql/) as defined in the NServiceBus [support policy](/nservicebus/upgrades/support-policy.md):
+
+ * [3.1.1](https://github.com/Particular/NServiceBus.SqlServer/releases/tag/3.1.1)
+ * [3.0.3](https://github.com/Particular/NServiceBus.SqlServer/releases/tag/3.0.3)
+ * [2.2.6](https://github.com/Particular/NServiceBus.SqlServer/releases/tag/2.2.6)
+
+Any of the supported affected minor versions (3.1.x, 3.0.x, or 2.2.x) should be updated to the latest patch release. Older (unsupported) affected versions should be updated to a newest supported minor (2.1.x or 2.0.x) or major version (1.x).
+
+
+## Upgrade steps
+
+To upgrade an existing endpoint:
+
+ * Update to the latest patch release
+ * Deploy the new version
+ * Check endpoint startup logs for a [warning message](/transports/upgrades/sqlserver-unicode-headers.md#check-at-startup)
+ * Follow the [queue table schema upgrade procedure](/transports/upgrades/sqlserver-unicode-headers.md#queue-table-schema-upgrade)
+
+
+## Check at startup
+
+The SQL Server transport detects incorrect definition of the `Headers` column and logs a warning:
+
+> Table [dbo].[SampleEndpoint] stores headers in a non Unicode-compatible column (varchar).
+>
+> This may lead to data loss when sending non-ASCII characters in headers. SQL Server transport 3.1 and newer can take advantage of the nvarchar column type for headers. Please change the column type in the database.
```

Comment:
That's the warning users will have in the log. We'd have to release a patch to change it.

@SzymonPobiega There are more problems with this warning message, it seems. Thoughts?